src/formula/variant.cpp

```diff
@@ -771,57 +602,19 @@ bool variant::operator!=(const variant& v) const
 
 bool variant::operator<=(const variant& v) const
 {
-	if(type_ != v.type_) {
-		if(type_ == TYPE_DECIMAL && v.type_ == TYPE_INT) {
+	if(type() != v.type()) {
+		if(is_decimal()  && v.is_int()) {
 			return as_decimal() <= v.as_decimal();
 		}
-		if(v.type_ == TYPE_DECIMAL && type_ == TYPE_INT) {
-			return as_decimal() <= v.as_decimal();
-		}
-
-		return type_ < v.type_;
-	}
 
-	switch(type_) {
-	case TYPE_NULL: {
-		return true;
-	}
-
-	case TYPE_STRING: {
-		return *string_ <= *v.string_;
-	}
-
-	case TYPE_INT: {
-		return int_value_ <= v.int_value_;
-	}
-
-	case TYPE_DECIMAL: {
-		return decimal_value_ <= v.decimal_value_;
-	}
-
-	case TYPE_LIST: {
-		for(size_t n = 0; n != num_elements() && n != v.num_elements(); ++n) {
-			if((*this)[n] < v[n]) {
-				return true;
-			} else if((*this)[n] > v[n]) {
-				return false;
-			}
+		if(v.is_decimal() && is_int()) {
+			return as_decimal() <= v.as_decimal();
 		}
 
-		return num_elements() <= v.num_elements();
+		return type() < v.type();
 	}
 
-	case TYPE_MAP: {
-		return *map_ <= *v.map_;
-	}
-
-	case TYPE_CALLABLE: {
-		return !v.callable_->less(callable_);
-	}
-	}
-
-	assert(false);
-	return false;
+	return value_ <= v.value_;
```

Comment:
Again, this is pointer comparison, not value comparison. Presumably you want `*value_ <= *v.value_`.